### DISCUSSION TITLE
Running startup scripts with LTSP.conf

### DISCUSSION THREAD

**Discussion by HottuDoku** [ltsp/ltsp#747] [Q&A] [answered]
Hi,

I've ran into issue with running startup scripts through ltsp.conf.
According to man pages I created a section with `POST_INIT`. Then section with client identification (tried IP, MAC even hostname) and `INCLUDE`.
```
...
[testscript]
POST_INIT_TESTSCRIPT="touch /tmp/test.txt"
...
[172.25.9.14]
INCLUDE=testscript
...
```
---Ran `ltsp initrd` here---

I have my own script i want to run, but for the sake of testing I used something simpler, however even this doesn't do anything. Is there any way to run these scripts for specific clients through ltsp.conf?

**Comment by alkisg** [comment]
Hi, did you run `ltsp initrd`? It's needed after every ltsp.conf change.

Additionally, try `touch /run/test.txt` instead, as in many distributions /tmp is cleared at boot.

**Reply by HottuDoku** [reply]
Yes, I did run `ltsp initrd` after every change.

So I tried what you suggested and it did work. There comes the second problem. I want to run a script that captures barcode scanner output through RS232 and pass it to registered application.
I tried to replace `POST_INIT` line with this:
`POST_INIT_BARCODEREADER="/opt/replicator/barcode -c 96 -b 9600 -p 3 -d 7 -s 2 -l 6700 & "`

So my configuration looks like this:
```
...
[barcodereader]
POST_INIT_BARCODEREADER="/opt/replicator/barcode -c 96 -b 9600 -p 3 -d 7 -s 2 -l 6700 & "
...
[172.25.9.14]
INCLUDE=barcodereader
...
```
When I run the script manualy its working. But it doesn't on boot through ltsp.

**Comment by alkisg** [comment, marked as answer]
Try to put it in POST_SERVICE_BARCODEREADER instead.
POST_INIT is too early for "daemons" as not even systemd is up yet.

And of course you could create a proper systemd unit instead of relying on LTSP to launch daemons.
